### DISCUSSION TITLE
Passing Detox Instance test event listeners

### DISCUSSION THREAD

**Discussion by iMaupin** [wix/Detox#3708] [Q&A]
Hello,

For my test suite, I have a Custom Circus Environment that passes a custom-built listener class that I use for Allure Reporting. Prior to v.20 all listeners were instantiated and passed the environment and a detox instance. I notice that in v20 this detox instance has been removed.

Is it possible that this could be readded? Or is there another workaround that you suggest to get access to the device in order to capture screenshots for Allure?

**Comment by portyu9** [comment]
Hi @iMaupin 

I would avoid depending on the old listener behavior that passed the internal Detox instance around. That was private runner integration behavior and changed in Detox 20.

If all you need is a screenshot, use the public `device` API from the initialized Detox worker:

```js
const { device } = require('detox')

const path = await device.takeScreenshot('my-screenshot')
```

So you should not need access to the entire internal Detox instance just to capture a screenshot.

For an Allure integration, another option is to let Detox's artifact system capture failure screenshots and then attach those artifacts to the Allure result.

If your custom runner integration genuinely needs lifecycle-level internals, use the current Detox runner/internals APIs rather than relying on the pre-v20 Jest Circus constructor/listener contract.

Please mark this answer as accepted if it helped, thank you.
